cache/fast_lru_cache.h

```diff
@@ -105,54 +131,96 @@ struct LRUHandle {
     assert(total_charge >= meta_charge);
     return total_charge - meta_charge;
   }
+
+  // A handle can represent 3 different things: an empty slot, a tombstone, or
+  // an actual element.
+  inline bool IsEmpty() {
+    return !this->IsElement() && this->displacements == 0;
+  }
+
+  inline bool IsTombstone() {
+    return !this->IsElement() && this->displacements > 0;
+  }
+
+  inline bool Matches(const Slice& some_key, uint32_t some_hash) {
```

Comment:
IsMatch ? Or another name like IsEmpty ?

Reply:
Bear in mind that IsEmpty, IsTombstone, IsElement and IsVisible check the state of a handle (see the code comment about the definition of "state"). So perhaps it's okay if the naming differs?